source-stripe-native/source_stripe_native/api.py

```diff
@@ -328,9 +333,21 @@ async def _capture_substreams(
     while True:
         if cls_child.NAME == "Persons" and parent_data.controller["requirement_collection"] == "stripe" :
             break
-        result_child = ListResult[cls_child].model_validate_json(
-            await http.request(log, child_url, method="GET", params=parameters)
-        )
+
+        try:
+            result_child = ListResult[cls_child].model_validate_json(
+                await http.request(log, child_url, method="GET", params=parameters)
+            )
+        except HTTPError as err:
+            # It's possible for us to process events for deleted parent resources, making
+            # the requests for the associated child resources fail. Stripe returns a 404
+            # error & a message containing "resource_missing" and "No such" when this happens.
+            if err.code == 404 and bool(re.search(MISSING_RESOURCE_REGEX, err.message, re.DOTALL)):
+                log.debug(f"Missing resource error for URL {child_url}. Skipping to the next resource.", err)
```

Comment:
IDK how common it is to get 404 responses, but it might be helpful to log at info or warn levels as long as it wouldn't be too spammy. Feel free to ignore this suggestion if it'd result in a bunch of log spam though.

Reply:
Thanks for the suggestion, Phil! I don't anticipate we'd encounter this error all that often. I'll switch it over to warn.